go/worker/registration/worker.go

```diff
@@ -769,7 +769,7 @@ func (w *Worker) newRoleProvider(role node.RolesMask, runtimeID *common.Namespac
 		"id", runtimeID,
 		"role", role,
 	)
-	if !role.IsSingleRole() {
+	if role != node.RoleEmpty && !role.IsSingleRole() {
 		return nil, fmt.Errorf("RegisterRole: registration role mask does not encode a single role. RoleMask: '%s'", role)
```

Comment:
The comment needs to be fixed.
What is a string representation of the empty role? Should it be `[no roles]`?

Reply:
Why does the error message need to be fixed? It would still be correct as it would never be emitted for the empty role.